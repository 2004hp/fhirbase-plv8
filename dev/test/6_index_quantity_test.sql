--db:fhirb -e
--{{{
\set pt `curl http://www.hl7.org/implement/standards/fhir/observation-example-f001-glucose.json`

WITH res AS (
  SELECT insert_resource(:'pt'::jsonb)
)
select count(*) from observation_search_quantity
where resource_id NOT IN (select insert_resource from res);
--}}}

--{{{
select count(*) from fhir.resource_indexables
where resource_type = 'Observation';
--}}}

--{{{
select count(*) from fhir.primitive_types
order by type
--}}}
--{{{
select count(*) from fhir.expanded_resource_elements
where
path[1] = 'Observation'
and path[2] = 'subject'
order by path ;
--}}}
