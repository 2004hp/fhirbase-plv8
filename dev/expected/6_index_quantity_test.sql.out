SELECT insert_resource( E'{"resourceType":"Observation","text":{"status":"generated","div":"<div>\\n      <p>\\n        <b>Generated Narrative</b>\\n      </p>\\n      <p>\\n        <b>name</b>: \\n        <span title=\\"Codes: {http://loinc.org 2339-0}\\">Glucose [Mass/volume] in Blood</span>\\n      </p>\\n      <p>\\n        <b>value[x]</b>: 6.3 mmol/l\\n      </p>\\n      <p>\\n        <b>interpretation</b>: \\n        <span title=\\"Codes: {http://hl7.org/fhir/v2/0078 A}\\">abnormal</span>\\n      </p>\\n      <p>\\n        <b>applies[x]</b>: 2-Apr 2013 9:30 --&gt; 5-Apr 2013 9:30\\n      </p>\\n      <p>\\n        <b>issued</b>: 3-Apr 2013 15:30\\n      </p>\\n      <p>\\n        <b>status</b>: final_\\n      </p>\\n      <p>\\n        <b>reliability</b>: ok\\n      </p>\\n      <p>\\n        <b>bodySite</b>: \\n        <span title=\\"Codes: {http://snomed.info/sct 308046002}\\">Superficial forearm vein</span>\\n      </p>\\n      <p>\\n        <b>method</b>: \\n        <span title=\\"Codes: {http://snomed.info/sct 120220003}\\">Injection to forearm</span>\\n      </p>\\n      <p>\\n        <b>identifier</b>: 6323 (official)\\n      </p>\\n      <p>\\n        <b>subject</b>: P. van de Heuvel\\n      </p>\\n      <p>\\n        <b>performer</b>: A. Langeveld\\n      </p>\\n      <h3>ReferenceRanges</h3>\\n      <table class=\\"grid\\">\\n        <tr>\\n          <td>\\n            <b>Low</b>\\n          </td>\\n          <td>\\n            <b>High</b>\\n          </td>\\n          <td>\\n            <b>Meaning</b>\\n          </td>\\n          <td>\\n            <b>Age</b>\\n          </td>\\n        </tr>\\n        <tr>\\n          <td>3.1 mmol/l</td>\\n          <td>6.2 mmol/l</td>\\n          <td> </td>\\n          <td> </td>\\n        </tr>\\n      </table>\\n    </div>"},"name":{"coding":[{"system":"http://loinc.org","code":"2339-0","display":"Glucose [Mass/volume] in Blood"}]},"valueQuantity":{"value":6.3,"units":"mmol/l","system":"http://unitsofmeasure.org","code":"mmol/l"},"interpretation":{"coding":[{"system":"http://hl7.org/fhir/v2/0078","code":"A","display":"abnormal"}]},"appliesPeriod":{"start":"2013-04-02T09:30:10+01:00","end":"2013-04-05T09:30:10+01:00"},"issued":"2013-04-03T15:30:10+01:00","status":"final","reliability":"ok","bodySite":{"coding":[{"system":"http://snomed.info/sct","code":"308046002","display":"Superficial forearm vein"}]},"method":{"coding":[{"system":"http://snomed.info/sct","code":"120220003","display":"Injection to forearm"}]},"identifier":{"use":"official","system":"http://www.bmc.nl/zorgportal/identifiers/observations","value":"6323"},"subject":{"reference":"Patient/f001","display":"P. van de Heuvel"},"performer":[{"reference":"Practitioner/f005","display":"A. Langeveld"}],"referenceRange":[{"low":{"value":3.1,"units":"mmol/l","system":"http://unitsofmeasure.org","code":"mmol/l"},"high":{"value":6.2,"units":"mmol/l","system":"http://unitsofmeasure.org","code":"mmol/l"}}]}'::jsonb);
           insert_resource            
--------------------------------------
 2120f48d-5549-40ac-b286-46b2b9e40b97
(1 row)

select * from observation_search_quantity;
 _id |             resource_id              |     param      | value | comparator | units  |          system           |  code  
-----+--------------------------------------+----------------+-------+------------+--------+---------------------------+--------
   1 | bb9b3c1b-b684-4610-810e-ed3acf148680 | value-quantity |   6.3 |            | mmol/l | http://unitsofmeasure.org | mmol/l
   2 | 2120f48d-5549-40ac-b286-46b2b9e40b97 | value-quantity |   6.3 |            | mmol/l | http://unitsofmeasure.org | mmol/l
(2 rows)

/* select * from fhir.resource_search_params */
/* where path[1] = 'Observation' */
/* ; */
select * from fhir.resource_indexables
where resource_type = 'Observation';
   param_name   | resource_type | search_type |       type        | max |                path                | is_primitive 
----------------+---------------+-------------+-------------------+-----+------------------------------------+--------------
 _id            | Observation   | token       | Resource          | 1   | {Observation}                      | f
 name-value-[x] | Observation   | composite   | Resource          | 1   | {Observation}                      | f
 related        | Observation   | composite   | Resource          | 1   | {Observation}                      | f
 date           | Observation   | date        | dateTime          | 1   | {Observation,appliesdateTime}      | f
 date           | Observation   | date        | Period            | 1   | {Observation,appliesPeriod}        | f
 name           | Observation   | token       | CodeableConcept   | 1   | {Observation,name}                 | f
 performer      | Observation   | reference   | ResourceReference | *   | {Observation,performer}            | f
 related-target | Observation   | reference   | ResourceReference | 1   | {Observation,related,target}       | f
 related-type   | Observation   | token       | code              | 1   | {Observation,related,type}         | t
 reliability    | Observation   | token       | code              | 1   | {Observation,reliability}          | t
 specimen       | Observation   | reference   | ResourceReference | 1   | {Observation,specimen}             | f
 status         | Observation   | token       | code              | 1   | {Observation,status}               | t
 subject        | Observation   | reference   | ResourceReference | 1   | {Observation,subject}              | f
 value-string   | Observation   | string      | Attachment        | 1   | {Observation,valueAttachment}      | f
 value-concept  | Observation   | token       | CodeableConcept   | 1   | {Observation,valueCodeableConcept} | f
 value-string   | Observation   | string      | CodeableConcept   | 1   | {Observation,valueCodeableConcept} | f
 value-date     | Observation   | date        | Period            | 1   | {Observation,valuePeriod}          | f
 value-string   | Observation   | string      | Period            | 1   | {Observation,valuePeriod}          | f
 value-quantity | Observation   | quantity    | Quantity          | 1   | {Observation,valueQuantity}        | f
 value-string   | Observation   | string      | Quantity          | 1   | {Observation,valueQuantity}        | f
 value-string   | Observation   | string      | Ratio             | 1   | {Observation,valueRatio}           | f
 value-string   | Observation   | string      | SampledData       | 1   | {Observation,valueSampledData}     | f
 value-concept  | Observation   | token       | string            | 1   | {Observation,valuestring}          | t
 value-string   | Observation   | string      | string            | 1   | {Observation,valuestring}          | t
(24 rows)

select * from fhir.primitive_types
order by type


select * from fhir.expanded_resource_elements
where
path[1] = 'Observation'
and path[2] = 'subject'
order by path
;
